Content.Server/Tabletop/TabletopSystem.cs

```diff
@@ -24,184 +20,76 @@ public partial class TabletopSystem : SharedTabletopSystem
         }
 
-        #region Event handlers
-
-        // Move an entity which is dragged by the user, but check if they are allowed to do so and to these coordinates
-        private void OnTabletopMove(TabletopMoveEvent msg, EntitySessionEventArgs args)
+        private void OnGamerShutdown(EntityUid uid, TabletopGamerComponent component, ComponentShutdown args)
         {
-            if (args.SenderSession as IPlayerSession is not { AttachedEntity: { } playerEntity } playerSession) return;
-
-            // Check if player is actually playing at this table
-            if (!_gameSessions.TryGetValue(msg.TableUid, out var tableUid) ||
-                !tableUid.IsPlaying(playerSession)) return;
+            if (!ComponentManager.TryGetComponent(uid, out ActorComponent? actor))
+                return;
 
-            // Return if can not see table or stunned/no hands
-            if (!EntityManager.TryGetEntity(msg.TableUid, out var table)) return;
-            if (!CanSeeTable(playerEntity, table) || StunnedOrNoHands(playerEntity)) return;
-
-            // Check if moved entity exists and has tabletop draggable component
-            if (!EntityManager.TryGetEntity(msg.MovedEntityUid, out var movedEntity)) return;
-            if (!ComponentManager.HasComponent<TabletopDraggableComponent>(movedEntity.Uid)) return;
-
-            // TODO: some permission system, disallow movement if you're not permitted to move the item
-
-            // Move the entity and dirty it (we use the map ID from the entity so noone can try to be funny and move the item to another map)
-            var transform = ComponentManager.GetComponent<ITransformComponent>(movedEntity.Uid);
-            var entityCoordinates = new EntityCoordinates(_mapManager.GetMapEntityId(transform.MapID), msg.Coordinates.Position);
-            transform.Coordinates = entityCoordinates;
-            movedEntity.Dirty();
+            if(component.Tabletop.IsValid())
+                CloseSessionFor(actor.PlayerSession, component.Tabletop);
         }
 
-        private void OnDraggingPlayerChanged(TabletopDraggingPlayerChangedEvent msg)
+        public override void Update(float frameTime)
         {
-            var draggedEntity = EntityManager.GetEntity(msg.DraggedEntityUid);
-
-            if (!draggedEntity.TryGetComponent<TabletopDraggableComponent>(out var draggableComponent)) return;
-
-            draggableComponent.DraggingPlayer = msg.DraggingPlayer;
+            base.Update(frameTime);
 
-            if (!draggedEntity.TryGetComponent<AppearanceComponent>(out var appearance)) return;
-
-            if (draggableComponent.DraggingPlayer != null)
-            {
-                appearance.SetData(TabletopItemVisuals.Scale, new Vector2(1.25f, 1.25f));
-                appearance.SetData(TabletopItemVisuals.DrawDepth, (int) DrawDepth.Items + 1);
-            }
-            else
+            foreach (var gamer in ComponentManager.EntityQuery<TabletopGamerComponent>())
```

Comment:
Should this include paused gamers? As by default it won't.

Reply:
yeah, it should